scapy/contrib/dtn/tcpcl_session.py

```diff
@@ -0,0 +1,133 @@
+# SPDX-License-Identifier: GPL-2.0-only
+# This file is part of Scapy
+# See https://scapy.net/ for more information
+
+# scapy.contrib.description = TCP Convergence Layer version 4 (TCPCLv4)
+# scapy.contrib.status = loads
+
+# These classes support unit testing of the TCPCL scapy layer
+# (scapy.contrib.dtn.tcpcl) and illustrate how the protocol messages may
+# be used to emulate a TCPCL session.
+
+from scapy.all import Raw, raw, TCP, Packet, bind_layers, split_layers
+import scapy.contrib.dtn.tcpcl as TCPCL
+from typing import List
+
+
+class Session:
```

Comment:
I'm pretty sure this could be implemented using Scapy's "session", which allows users to have their own `prn` while still performing actions when receiving a packet in a `sniff`, `rdpcap` or other packet source.

There's a bit of documentation here 
https://scapy.readthedocs.io/en/latest/usage.html#advanced-sniffing-sniffing-sessions
and you can have a look at the source https://github.com/secdev/scapy/blob/master/scapy/sessions.py

Typically you can extend `IPSession` if you want it to handle IP fragmentation, or `TCPSession` (with a few tricks) to handle TCP reassembly